src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs

```diff
@@ -0,0 +1,582 @@
+                CancellationToken cancellationToken)
+            {
+                var syntaxFacts = inlineMethodRefactoringProvider._syntaxFacts;
+                var semanticFactsService = inlineMethodRefactoringProvider._semanticFactsService;
+                var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var syntaxGenerator = SyntaxGenerator.GetGenerator(document);
+                var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var inlineNode = inlineMethodRefactoringProvider.GetInlineStatement(calleeMethodDeclarationNode);
+
+                // Get the parameterSymbol to identifier's name mappings.
+                // Example:
+                // void Caller()
+                // {
+                //     int i = 10;
+                //     Callee(i);
+                // }
+                // void Callee(int a)
+                // {
+                //     DoSomething(a);
+                // }
+                // Input is an array of (ParameterSymbol(int a), IdentifierSyntaxNode(int i))
+                // Output is an array of (ParameterSymbol(int a), string i)
+                var parametersWithIdentifierArgumentToName = methodParametersInfo.ParametersWithIdentifierArgument
+                    .Select(parameterAndArgument =>
+                        (parameterAndArgument.parameterSymbol, semanticModel.GetSymbolInfo(parameterAndArgument.identifierSyntaxNode, cancellationToken).GetAnySymbol()?.Name))
+                    .Where(parameterAndName => parameterAndName.Name != null)
+                    .ToImmutableArray();
+
+                // Get the parameterSymbol to variable declaration name mapping.
+                // Example:
+                // void Caller()
+                // {
+                //     Callee(out int i);
+                // }
+                // void Callee(out int a)
+                // {
+                //     DoSomething(out a);
+                // }
+                // Input is an array of (ParameterSymbol(int a), IdentifierSyntaxNode(int i))
+                // Output is an array of (ParameterSymbol(int a), string i)
+                var parametersWithVariableDeclarationArgumentToName = methodParametersInfo
+                    .ParametersWithVariableDeclarationArgument
+                    .Select(parameterAndArgument =>
+                        (parameterAndArgument.parameterSymbol,
+                            semanticModel
+                                .GetSymbolInfo(parameterAndArgument.variableDeclarationNode, cancellationToken)
+                                .GetAnySymbol()?.Name))
+                    .Where(parameterAndName => parameterAndName.Name != null)
+                    .ToImmutableArray();
+
+                // Based on the information get above, generate a map which the key is the symbol need renaming, value is the new name.
```

Comment:
needs an example.  i.e. when is renamign needed.

Reply:
I think I put all the example in the method. Do you mean I should move them to here?